src/App.tsx

```diff
@@ -1,8 +1,69 @@
 import React from 'react';
 import './App.css';
+import { GoodsList } from './components/GoodsList';
+import { getGoods } from './api/api';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface State {
+  goods: Good[];
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [],
+  };
+
+  loadGoods = () => {
+    getGoods().then(goods => this.setState({
+      goods,
+    }));
+  };
+
+  loadFiveGoods = () => {
+    getGoods().then(goods => this.setState({
+      goods: goods.sort((a: Good, b: Good) => a.name.localeCompare(b.name))
```

Comment:
если getGoods задать корректные типы return, то тут не нужно 
```suggestion
      goods: goods.sort((a, b) => a.name.localeCompare(b.name))
```